sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveShim.scala

```diff
@@ -196,14 +194,8 @@ private[hive] object HiveShim {
       if (instance != null) {
         instance.asInstanceOf[UDFType]
       } else {
-        val func = Utils.getContextOrSparkClassLoader
+        Utils.getContextOrSparkClassLoader
           .loadClass(functionClassName).newInstance.asInstanceOf[UDFType]
-        if (!func.isInstanceOf[UDF]) {
```

Comment:
`instance` is now not null just for macro function, which contains detail of how to handle input data. I've removed caching UDF because hive udf does not coerces functions to be stateless, afaik. And I don't think creating UDF instance will make any performance regression.